ext/geos_c_impl/factory.c

```diff
@@ -832,7 +832,8 @@ rgeo_convert_to_geos_geometry(VALUE factory, VALUE obj, VALUE type)
       rb_funcall(rgeo_feature_module, rb_intern("cast"), 3, obj, factory, type);
   }
   if (NIL_P(object))
-    return NULL;
+    rb_raise(rb_eRGeoInvalidGeometry,
+             "Unable to cast the geometry to the GEOS Factory");
```

Comment:
Some tests were failing because they check that an `InvalidGeometry` error is raised rather than any error, so I left the conditional in for now. We can remove it all together and allow `Check_TypedStruct` to raise, but we'll need to fix the tests.